system/Image/GD.php

```diff
@@ -0,0 +1,199 @@
+<?php namespace CodeIgniter\Image;
+
+/**
+ * CodeIgniter
+ *
+ * An open source application development framework for PHP
+ *
+ * This content is released under the MIT License (MIT)
+ *
+ * Copyright (c) 2014 - 2016, British Columbia Institute of Technology
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @package	CodeIgniter
+ * @author	CodeIgniter Dev Team
+ * @copyright	Copyright (c) 2014 - 2016, British Columbia Institute of Technology (http://bcit.ca/)
+ * @license	http://opensource.org/licenses/MIT	MIT License
+ * @link	http://codeigniter.com
+ * @since	Version 3.0.0
+ * @filesource
+ */
+
+/**
+ * Image manipulation GD API.
+ *
+ * @package CodeIgniter\Image
+ */
+class GD extends Image implements ImageInterface
+{
+    /**
+     * Function to resize image.
+     * 
+     * @param resource $image
+     * @param int $new_width
+     * @param int $new_height
+     * @return resource
+     */
+    public function resize(int $new_width, int $new_height): resource
+    {
+        imagescale($this->resource, $new_width, $new_height);
+        
+        return $this->resource;
+    }
+    
+    // ------------------------------------------------------------------
+    
+    /**
+     * {@inheritDoc}
+     * @see \CodeIgniter\Image\ImageInterface::crop()
+     */
+    public function crop(int $x, int $y, int $new_width, int $new_height): resource
+    {
+        imagecrop($this->resource, ['x' => $x, 'y' => $y,
+                        'width' => $new_width, 'height' => $new_height]);
+        
+        return $this->resource;
+    }
+    
+    // ------------------------------------------------------------------
+    
+    /**
+     * {@inheritDoc}
+     * @see \CodeIgniter\Image\ImageInterface::rotate()
+     */
+    public function rotate(int $angle, string $bg_color = 'fff', string $transparency = 0): resource
+    {
+        imagerotate($this->resource, $angle, $bgd_color, $transparency);
```

Comment:
Seems like we should be providing some form of validation of the validity of the parameters here to ensure they're within valid ranges and provide decent error messages, unless the error messages we get back from failures of that command are already good?